sympy/assumptions/handlers/matrices.py

```diff
@@ -228,46 +273,54 @@ def MatPow(expr, assumptions):
         return None
 
     @staticmethod
-    def MatAdd(expr, assumptions):
+    def MatAdd(expr, *, assumptions):
         if (len(expr.args) == 1 and
                 ask(Q.orthogonal(expr.args[0]), assumptions)):
             return True
 
     @staticmethod
-    def MatrixSymbol(expr, assumptions):
+    def MatrixSymbol(expr, *, assumptions):
         if (not expr.is_square or
                         ask(Q.invertible(expr), assumptions) is False):
             return False
         if Q.orthogonal(expr) in conjuncts(assumptions):
             return True
 
-    Identity = staticmethod(CommonHandler.AlwaysTrue)
-
-    ZeroMatrix = staticmethod(CommonHandler.AlwaysFalse)
-
     @staticmethod
-    def Transpose(expr, assumptions):
+    def Transpose(expr, *, assumptions):
         return ask(Q.orthogonal(expr.arg), assumptions)
 
-    Inverse = Transpose
-
     @staticmethod
-    def MatrixSlice(expr, assumptions):
+    def MatrixSlice(expr, *, assumptions):
         if not expr.on_diag:
             return None
         else:
             return ask(Q.orthogonal(expr.parent), assumptions)
 
-    Factorization = staticmethod(partial(_Factorization, Q.orthogonal))
+AskOrthogonalHandler = AskOrthogonalHandlerClass('AskOrthogonalHandler')
+
 
-class AskUnitaryHandler(CommonHandler):
+class AskUnitaryHandlerClass(CommonHandlerClass):
     """
     Handler for key 'unitary'
     """
-    predicate = Q.unitary
+
+    def __init__(self, name, doc=None):
+        super().__init__(name, doc)
+
+        for sig in (
+        MatMul, MatPow, MatrixSymbol, MatrixSlice
+        ):
+            self.register(sig)(getattr(self, sig.__name__))
```

Comment:
This seems to be a common idiom here. Is there a better way of doing this?

I do wonder whether it would make more sense to define these handlers in the matrices modue rather than here in assumptions.

Reply:
Perhaps there could be a decorator to indicate which classes a handler method should be registered with like:
```python
    @register(MatPow, MatMul):
    def MatMul():
        ....
```

Reply:
Decorator seems better. I will try this way.

> I do wonder whether it would make more sense to define these handlers in the matrices modue rather than here in assumptions.

IMO defining in assumtions/handlers makes them more easy to be found.

Reply:
I modified this in other way. Instead of keeping the classes, I made multipledispatch handlers copyable. Now, for example, `AskExtendedRealHandler` and `AskRealHandler` are both instances of `AskHandlerClass` instead of having each one's own class. It is just that former is copied from the latter and dispatched with some more functions.